tests/admin_widgets/tests.py

```diff
@@ -1781,6 +1818,63 @@ def test_ForeignKey_using_to_field(self):
         self.assertEqual(profiles[0].user.username, username_value)
 
 
+class RelatedFieldWidgetSeleniumPrimaryKeyTests(AdminWidgetSeleniumTestCase):
+    def test_create_houses_and_room(self):
+        from selenium.webdriver.common.by import By
+        from selenium.webdriver.support.select import Select
+
+        self.admin_login(username="super", password="secret", login_url="/")
+
+        # Create a new House with a PK that needs quoting
+        house = House.objects.create(name="_40")
+        house_name = str(house.name)
+        self.selenium.get(
+            self.live_server_url + reverse("admin:admin_widgets_house_add")
+        )
+
+        self.selenium.find_element(By.ID, "id_name").send_keys(house_name)
+        self.selenium.find_element(By.NAME, "_continue").click()
+
+        # Check that the new House was created and listed in the change form
+        self.selenium.get(
+            self.live_server_url + reverse("admin:admin_widgets_room_add")
+        )
+        self.assertIn(house_name, self.selenium.page_source)
+
+        # Find and interact with the select element for choosing a House
+        select_house = Select(self.selenium.find_element(By.ID, "id_house"))
+        select_house.select_by_index(0)
+
+        select_house.select_by_value(house_name)
+        self.assertEqual(
+            select_house.first_selected_option.get_attribute("value"), house_name
+        )
+
+        # Create a new Room and associate it with the House created above
+        room = Room.objects.create(name="onebed", house=house)
+        room_name = str(room.name)
+        self.selenium.find_element(By.ID, "id_name").send_keys(room_name)
+
+        self.selenium.find_element(By.NAME, "_continue").click()
+
+        save_button_css_selector = ".submit-row > input[type=submit]"
+
+        self.selenium.find_element(By.CSS_SELECTOR, save_button_css_selector)
+        self.selenium.find_element(By.ID, "view_id_house").click()
+        self.selenium.back()
+        time.sleep(60)
```

Comment:
This sleep is huuge :smile: Do we need it?

Reply:
> This sleep is huuge 😄 Do we need it?
The sleep is not needed, I added it when i was trying to inspect the browser so the selenium test doesn't end. I thought I removed it but I will remove it. I will definitely work on the comments and blank spaces

Reply:
The sleep is not needed, I added it when i was trying to inspect the browser so the selenium doesn't end. I thought i removed it but I will remove it.